modules/Oculus_module/src/OculusModule.cpp

```diff
@@ -967,19 +1240,12 @@ bool OculusModule::updateModule()
 
         // prepare robot (A button)
         m_joypadControllerInterface->getButton(m_prepareWalkingIndex, buttonMapping);
-        yarp::os::Bottle cmd, outcome;
-
         if (buttonMapping > 0)
         {
-            // TODO add a visual feedback for the user
-            if (m_moveRobot)
-            {
-                cmd.addString("prepareRobot");
-                m_rpcWalkingClient.write(cmd, outcome);
-            }
-            m_state = OculusFSM::InPreparation;
-            yInfo() << "[OculusModule::updateModule] prepare the robot";
+            std::lock_guard<std::mutex> guard(m_mutex);
```

Comment:
The lock should not just be here. For example, in the lines above you are accessing to the ``m_state`` variable, that can be edited at anytime by the RPC calls. The scope of the lock should be on the entire ``updateModule``.